### PR TITLE
KEYCLOAK-16923 Revising Authorization Services Guide for Red Hat Standards

### PR DESCRIPTION
**Pull request by andymunro**
@pdrozd This is the last guide in the documentation, which is now revised to meet Red Hat standards for findability and reusability of topics.

### CODE REVIEW

File: authorization_services/topics/hello-world-create-realm.adoc [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
 
-. On the right side of the empty user list, click *Add User*.
+. Toggle *User Enabled* to *ON*
 
-. To create a new user, complete the *Username*, *Email*, *First Name*, and *Last Name* fields.
-Click the *User Enabled* switch to *On*, and then click *Save*.
+. Click *Save*.
 +
 .Add User
 image:{project_images}/getting-started/hello-world/create-user.png[alt="Add User"]
```
Comment (nehachopra27) [inline authorization_services/topics/hello-world-create-realm.adoc:26]: @andymunro : The casing of image heading is all caps throughout this guide. Does it needs to match the other headings? Please check and if required then it needs to be changed throughout.

Comment (andymunro) [reply authorization_services/topics/hello-world-create-realm.adoc:26]: Thanks.  You are right, @nehachopra27.  I will change it to sentence case, meaning lower case except for the first word or a proper name.

File: authorization_services/topics/policy-evaluation-api.adoc [hunk 2]
```diff
@@ -54,7 +54,7 @@ Policies determine this by invoking the `grant()` or `deny()` methods on an `Eva
 
 For more information about the Evaluation API see the {apidocs_link}[JavaDocs].
```
Comment (nehachopra27) [inline authorization_services/topics/policy-evaluation-api.adoc]: @andymunro: Should this be changed to Additional Resources?

File: authorization_services/topics/permission-create-resource.adoc [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-= Creating Resource-Based Permissions
+= Creating resource-based permission
 
 A resource-based permission defines a set of one or more resources to protect using a set of one or more authorization policies.
 
-To create a new resource-based permission, select *Resource-based* in the dropdown list in the upper right corner of the permission listing.
+To create a new resource-based permission, select *Resource-based* in the item list in the upper right corner of the permission listing.
 
 .Add Resource-Based Permission
 image:{project_images}/permission/create-resource.png[alt="Add Resource-Based Permission"]
```
Comment (nehachopra27) [inline authorization_services/topics/permission-create-resource.adoc]: Image casing

File: authorization_services/topics/permission-create-scope.adoc [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
-= Creating Scope-Based Permissions
+= Creating scope-based permissions
 
 A scope-based permission defines a set of one or more scopes to protect using a set of one or more authorization policies. Unlike resource-based permissions, you can use this permission type to create permissions not only for a resource, but also for the scopes associated with it, providing more granularity when defining the permissions that govern your resources and the actions that can be performed on them.
 
-To create a new scope-based permission, select *Scope-based* in the dropdown list in the upper right corner of the permission listing.
+To create a new scope-based permission, select *Scope-based* in the item list in the upper right corner of the permission listing.
 
 .Add Scope-Based Permission
 image:{project_images}/permission/create-scope.png[alt="Add Scope-Based Permission"]
```
Comment (nehachopra27) [inline authorization_services/topics/permission-create-scope.adoc]: image title casing

File: authorization_services/topics/policy-aggregated-policy.adoc [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
-= Aggregated Policy
+= Aggregated policy
 
 As mentioned previously, {project_name} allows you to build a policy of policies, a concept referred to as policy aggregation. You can use policy aggregation to reuse existing policies to build more complex ones and keep your permissions even more decoupled from the policies that are evaluated during the processing of authorization requests.
 
-To create a new aggregated policy, select *Aggregated* in the dropdown list located in the right upper corner of the policy listing.
+To create a new aggregated policy, select *Aggregated* in the item list located in the right upper corner of the policy listing.
 
 .Add an Aggregated Policy
 image:{project_images}/policy/create-aggregated.png[alt="Add Aggregated Policy"]
```
Comment (nehachopra27) [inline authorization_services/topics/policy-aggregated-policy.adoc]: image title casing

File: authorization_services/topics/policy-client-policy.adoc [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
-= Client-Based Policy
+= Client-based policy
 
 You can use this type of policy to define conditions for your permissions where a set of one or more clients is permitted to access an object.
 
-To create a new client-based policy, select *Client* in the dropdown list in the upper right corner of the policy listing.
+To create a new client-based policy, select *Client* in the item list in the upper right corner of the policy listing.
 
 .Add a Client-Based Policy
 image:images/policy/create-client.png[alt="Add Client-Based Policy"]
```
Comment (nehachopra27) [inline authorization_services/topics/policy-client-policy.adoc]: image title casing

File: authorization_services/topics/policy-client-scope-policy.adoc [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
 
 You can use this type of policy to define conditions for your permissions where a set of one or more client scopes is permitted to access an object.
 
 By default, client scopes added to this policy are not specified as required and the policy will grant access if the client requesting access has been granted any of these client scopes. However, you can specify a specific client scope as <<_policy_client_scope_required, required>> if you want to enforce a specific client scope.
 
-To create a new client scope-based policy, select *Client Scope* in the dropdown list in the upper right corner of the policy listing.
+To create a new client scope-based policy, select *Client Scope* in the item list in the upper right corner of the policy listing.
 
 .Add Client Scope-Based Policy
 image:images/policy/create-client-scope.png[alt="Add Client Scope-Based Policy"]
```
Comment (nehachopra27) [inline authorization_services/topics/policy-client-scope-policy.adoc]: image title casing

File: authorization_services/topics/policy-role-policy.adoc [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
 
 Role policies can be useful when you need more restricted role-based access control (RBAC), where specific roles must be enforced to grant access to an object. For instance, you can enforce that a user must consent to allowing a client application (which is acting on the user's behalf) to access the user's resources. You can use {project_name} Client Scope Mapping to enable consent pages or even enforce clients to explicitly provide a scope when obtaining access tokens from a {project_name} server.
 
-To create a new role-based policy, select *Role* in the dropdown list in the upper right corner of the policy listing.
+To create a new role-based policy, select *Role* in the item list in the upper right corner of the policy listing.
 
 .Add Role-Based Policy
 image:{project_images}/policy/create-role.png[alt="Add Role-Based Policy"]
```
Comment (nehachopra27) [inline authorization_services/topics/policy-role-policy.adoc]: image title casing

File: authorization_services/topics/hello-world-create-resource-server.adoc [hunk 9]
```diff
@@ -1,29 +1,40 @@
...
 
-To create a new client, complete the following steps:
+.Procedure
 
-. Click *Clients* to start creating a new client application and fill in the *Client ID*, *Client Protocol*, and *Root URL* fields.
+. Click *Clients* to start creating a client application.
+
+. Fill in the *Client ID*, *Client Protocol*, and *Root URL* fields.
 +
 .Create Client Application
```
Comment (nehachopra27) [inline authorization_services/topics/hello-world-create-resource-server.adoc]: image title casing here is not read from alt text, it is reading the . title.

File: authorization_services/topics/policy-client-scope-policy-required-client-scope.adoc [hunk 10]
```diff
@@ -8,4 +8,4 @@ image:images/policy/create-client-scope.png[alt="Example of Required Client Scop
```
Comment (nehachopra27) [inline authorization_services/topics/policy-client-scope-policy-required-client-scope.adoc:8]: Image title casing

File: authorization_services/topics/policy-group-policy.adoc [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
+= Group-based policy
 
 You can use this type of policy to define conditions for your permissions where a set of one or more groups (and their hierarchies) is permitted to access an object.
 
-To create a new group-based policy, select *Group* in the dropdown list in the upper right corner of the policy listing.
+To create a new group-based policy, select *Group* in the item list in the upper right corner of the policy listing.
 
-.Add Group-Based Policy
-image:{project_images}/policy/create-group.png[alt="Add Group-Based Policy"]
+Group-Based Policy
```
Comment (nehachopra27) [inline authorization_services/topics/policy-group-policy.adoc]: Image title casing

File: authorization_services/topics/policy-user-policy.adoc [hunk 12]
```diff
@@ -1,9 +1,9 @@
...
-= User-Based Policy
+= User-based policy
 
 You can use this type of policy to define conditions for your permissions where a set of one or more users is permitted to access an object.
 
-To create a new user-based policy, select *User* in the dropdown list in the upper right corner of the policy listing.
+To create a new user-based policy, select *User* in the item list in the upper right corner of the policy listing.
 
 .Add a User-Based Policy
 image:{project_images}/policy/create-user.png[alt="Add User-Based Policy"]
```
Comment (nehachopra27) [inline authorization_services/topics/policy-user-policy.adoc]: Image title casing

File: server_admin/topics/realms/keys.adoc [hunk 13]
```diff
@@ -17,6 +17,10 @@ When a realm is created a key pair and a self-signed certificate is automaticall
 . Click *Passive* to view passive keys.
 . Click *Disabled* to view disabled keys.
 
+<<<<<<< HEAD
```
Comment (nehachopra27) [inline server_admin/topics/realms/keys.adoc]: Conflict has been pushed here.

File: server_admin/topics/realms/keys.adoc [hunk 14]
```diff
@@ -37,6 +41,7 @@ Rotating keys also applies to offline tokens. To make sure they are updated, the
 
 ==== Adding a generated keypair
 
+<<<<<<< HEAD
```
Comment (nehachopra27) [inline server_admin/topics/realms/keys.adoc]: PLease check the conflict here

**Review by nehachopra27 — CHANGES_REQUESTED** [review]

File: authorization_services/topics/permission-create-resource.adoc (hunk 3 above)
Comment (andymunro) [reply authorization_services/topics/permission-create-resource.adoc]: I think this one should remain initial capitals to match the UI.

File: authorization_services/topics/permission-create-scope.adoc (hunk 4 above)
Comment (andymunro) [reply authorization_services/topics/permission-create-scope.adoc]: I think this one should remain initial capitals to match the UI.

File: authorization_services/topics/policy-aggregated-policy.adoc (hunk 5 above)
Comment (andymunro) [reply authorization_services/topics/policy-aggregated-policy.adoc]: I think this one should remain initial capitals to match the UI.

File: authorization_services/topics/policy-client-policy.adoc (hunk 6 above)
Comment (andymunro) [reply authorization_services/topics/policy-client-policy.adoc]: I think this one should remain initial capitals to match the UI.

File: authorization_services/topics/policy-client-scope-policy.adoc (hunk 7 above)
Comment (andymunro) [reply authorization_services/topics/policy-client-scope-policy.adoc]: I think this one should remain initial capitals to match the UI.

File: authorization_services/topics/policy-group-policy.adoc (hunk 11 above)
Comment (andymunro) [reply authorization_services/topics/policy-group-policy.adoc]: I think this one should remain initial capitals to match the UI.

File: authorization_services/topics/policy-user-policy.adoc (hunk 12 above)
Comment (andymunro) [reply authorization_services/topics/policy-user-policy.adoc]: I think this one should remain initial capitals to match the UI.

**Comment by andymunro** [comment]
@nehachopra27 Can you check this again.  There are still build errors about a missing "context" variable. I will ask development about these errors.  However, I addressed your feedback. Thanks.

**Comment by andymunro** [comment]
@nehachopra27  Any more input on this guide?

File: authorization_services/topics/policy-client-scope-policy-required-client-scope.adoc (hunk 10 above)
Comment (nehachopra27) [reply authorization_services/topics/policy-client-scope-policy-required-client-scope.adoc:8]: This casing has not been updated yet.

**Comment by andymunro** [comment]
@nehachopra27 I improved the use of sentence case which allows for initial capital letters to match the names on the screen.  It is hard to be consistent completely. This is probably the best I can do at this release.

File: authorization_services/topics/policy-role-policy.adoc [hunk 15]
```diff
@@ -33,4 +33,8 @@ Specifies which *client* roles are permitted by this policy. To enable this fiel
 +
 * *Logic*
 +
-The <<_policy_logic, Logic>> of this policy to apply after the other conditions have been evaluated.
+The logic of this policy to apply after the other conditions have been evaluated.
+
+[role="_additional-resources"]
+.Additional resources
+* <<_policy_logic, Postive and negative logic>>
```
Comment (nehachopra27) [inline authorization_services/topics/policy-role-policy.adoc]: @andymunro: Positive is misspelled here.

File: authorization_services/topics/policy-time-policy.adoc [hunk 16]
```diff
@@ -50,6 +50,10 @@ Defines the minute that access must be granted. You can also specify a range of
...
 * *Logic*
 +
-The <<_policy_logic, Logic>> of this policy to apply after the other conditions have been evaluated.
+The logic of this policy to apply after the other conditions have been evaluated.
 
 Access is only granted if all conditions are satisfied. {project_name} will perform an _AND_ based on the outcome of each condition.
+
+[role="_additional-resources"]
+.Additional resources
+* <<_policy_logic, Postive and negative logic>>
```
Comment (nehachopra27) [inline authorization_services/topics/policy-time-policy.adoc]: Positive is misspelled here.

File: authorization_services/topics/policy-user-policy.adoc [hunk 17]
```diff
@@ -25,4 +25,8 @@ Specifies which users are given access by this policy.
 +
 * *Logic*
 +
-The <<_policy_logic, Logic>> of this policy to apply after the other conditions have been evaluated.
+The logic of this policy to apply after the other conditions have been evaluated.
+
+[role="_additional-resources"]
+.Additional resources
+* <<_policy_logic, Postive and negative logic>>
```
Comment (nehachopra27) [inline authorization_services/topics/policy-user-policy.adoc]: Positive is misspelled here.

Comment (nehachopra27) [reply authorization_services/topics/policy-user-policy.adoc]: This is misspelled in all the subsection policies.

**Comment by nehachopra27** [comment]
@andymunro : I reviewed the changes. I had one additional comment which I raised, rest everything looks good.

**Comment by andymunro** [comment]
@nehachopra27 I made that correction. Thanks.  I think the errors reported are invalid.  The link identified as broken does take you to the Instagram topic in the Server Administration.

**Comment by nehachopra27** [comment]
@pdrozd : This guide is verified.

**Comment by andymunro** [comment]
@mposolda Would you be able to review this guide, which is approved by QE?

**Review by abstractj — APPROVED** [review]
LGTM

**Review by abstractj — CHANGES_REQUESTED** [review]
@andymunro that was too fast. Could you please address the failures below?

```
Error:  Failures: 
Error:    GuideTest.checkVariables:18->checkFailures:51 Variables not found  (1):
		- context
```

//cc @nehachopra27

**Comment by andymunro** [comment]
Thanks, @abstractj.  Yes, I have been trying to locate the cause of this failure.  The log file shows all the books built with no errors.

**Comment by abstractj** [comment]
@andymunro I noticed that our [admin server guide](https://raw.githubusercontent.com/keycloak/keycloak-documentation/21d73659caa9b2356f15bd253a9078a922cfb159/server_admin/master.adoc) uses a similar approach, the only difference is:

### Admin server guide

```:context: server_administration_guide``` - No colon

### Changes here

```:context: authorization_services_guide:``` - With colon at the end

I'm no expert on this, but maybe that's the issue.

**Comment by andymunro** [comment]
Thanks, @abstractj.  I tried removing the colon and then even removing the "context" code, however it still generates this error.  The product version does not have this error and the only difference in the books is the master.adoc versus index.adoc.  Maybe @pskopek has an idea.

**Comment by andymunro** [comment]
This PR is replace by this one: https://github.com/keycloak/keycloak-documentation/pull/1358

**Pull request closed without merging**